docs/src/pages/migration-guide.md

```diff
@@ -0,0 +1,221 @@
+  }
+})
+```
+
+To learn more about configuring Vite, please visit their [configuration guide](https://vitejs.dev/config/).
+
+
+
+## Aliases
+
+In Astro v0.21, import aliases can be added using `tsconfig.json` or `jsconfig.json`.
+
+```json
+{
+  "compilerOptions": {
+    "baseUrl": ".",
+    "paths": {
+      "@/components/*": ["components/*"]
+    }
+  }
+}
+```
+
+These aliases are integrated automatically into [VSCode](https://code.visualstudio.com/docs/languages/jsconfig) and other editors.
+
+
+
+## Aliases (#1749)
+
+In Astro v0.21, import aliases can also be added using the `alias` property.
+
+```js
+export default {
+  vite: {
+    resolve: {
+      alias: {
+        @components: '/src/components',
+      }
+    }
+  }
+}
+```
+
+Note: Always start alias values with `/`. Relative alias starting with `.` will be used as-is and will not be resolved.
+
+
+
+## Passing variables into scripts and styles
+
+In Astro v0.21, variables can be passed from Frontmatter into inline `<style>` and `<script>`.
+
+```astro
+---
+// tick.astro
+import { green } from '@radix-ui/colors'
+---
+<style define:vars={green}>
+  h-tick {
+    background-color: var(--green4);
+    border-radius: 50%;
+    color: var(--green11);
+    height: 15px;
+    width: 15px;
+  }
+</style>
+<h-tick>✓</h-tick>
+```
+
+
+
+## Components in Markdown
+
+In Astro v0.21, Components can be used within Markdown files.
+
+```md
+---
+Layout: '...'
+setup: | 
+  import MyReactComponent from '../components/MyReactComponent.jsx'
+---
+
+# Hydrating on visibility
+
+<MyReactComponent client:visible />
+```
+
+
+
+## Environment variables
+
+In Astro v0.21, environment variables are loaded from `.env` files in your project directory.
+
+```ini
+.env                # loaded in all cases
+.env.local          # loaded in all cases, ignored by git
+.env.[mode]         # only loaded in specified mode
+.env.[mode].local   # only loaded in specified mode, ignored by git
+```
+
+For security purposes, only variables prefixed with `PUBLIC_` are accessible to your code.
+In prior releases, these variables were prefixed with `SNOWPACK_PUBLIC_` and required the `@snowpack/plugin-env` plugin.
+
+**Example**
+
+```ini
+SECRET_PASSWORD=password123
+VITE_ANYBODY=there
+```
+
+In that example, only `VITE_ANYBODY` will be exposed as `import.meta.env.VITE_ANYBODY` to your client source code, but `SECRET_PASSWORD` will not.
+
+
+
+## File references
+
+In Astro v0.21, files need to be referenced with their extension exactly as it is on disk.
+
+This means that `file.tsx` files need to be referenced as `file.tsx` and not as `file.jsx`.
+
+**Example**
+
+```tsx
+// Div.tsx
+export default function Div(props) {
+  return <div />
+}
+```
+
+```diff
+- import Div from './Div.jsx' // Astro v0.20
++ import Div from './Div.tsx' // Astro v0.21
+```
+
+```scss
+// Div.scss
+div {
+  all: unset
+}
+```
+
+```diff
+- <link rel="stylesheet" href="./Div.css">
++ <link rel="stylesheet" href="./Div.scss">
```

Comment:
(Hopefully) resolved in 33e4d895e0134cc472a4ae0e61e88699beab8304. Would you take a look? I updated both to use `Astro.resolve` because updating just one felt more confusing to me to read.